### PR TITLE
Fixing bug in backward-in-time-writing

### LINKED ISSUE #1722: Backward in time outputs not written at desired frequency
### Parcels version

3.0.5

### Description

Looks like a regression from 3.0.4 to 3.0.5.

### Code sample

```python
from datetime import timedelta
import xarray as xr
import numpy as np
import parcels

example_dataset_folder = parcels.download_example_dataset("MovingEddies_data")
fieldset = parcels.FieldSet.from_parcels(f"{example_dataset_folder}/moving_eddies")

pset = parcels.ParticleSet.from_list(
    fieldset=fieldset,
    pclass=parcels.JITParticle,
    lon=[3.3e5, 3.3e5],
    lat=[1e5, 2.8e5],
)

# run forward

output_file = pset.ParticleFile(
    name="EddyParticles.zarr",
    outputdt=timedelta(hours=1),
)
pset.execute(
    parcels.AdvectionRK4,
    runtime=timedelta(days=2),
    dt=timedelta(minutes=5),
    output_file=output_file,
)

# run backward

output_file = pset.ParticleFile(
    name="EddyParticles_Bwd.zarr",
    outputdt=timedelta(hours=1),
)
pset.execute(
    parcels.AdvectionRK4,
    runtime=timedelta(days=2),
    dt=-timedelta(minutes=5),
    output_file=output_file,
)

# check that timesteps match (up to a sign)

ds = xr.open_zarr("EddyParticles.zarr/")
ds_bwd = xr.open_zarr("EddyParticles_Bwd.zarr/")

np.testing.assert_almost_equal(
    ds.time.isel(trajectory=0).diff("obs").mean().compute().astype(float).data[()],
    - ds_bwd.time.isel(trajectory=0).diff("obs").mean().compute().astype(float).data[()],
)
```

With `v3.0.5`:
```text
AssertionError: 
Arrays are not almost equal to 7 decimals
 ACTUAL: 3600000000000.0
 DESIRED: 86400000000000.0
```

With `v3.0.4` it works.

### PR DESCRIPTION
**Pull request by erikvansebille**
This fixes #1722, by reverting a change in ed11e43

Thanks for the git-bisect hint, @VeckoTheGecko!

### CODE REVIEW

File: tests/test_particlesets.py [hunk 1]
```diff
@@ -373,6 +385,73 @@ def AddLat(particle, fieldset, time):
...
+    file_outputdt = ds.isel(trajectory=0).time.diff(dim="obs").values
+    assert np.all(file_outputdt == np.timedelta64(-outputdt))
+
+
+def test_pset_execute_outputdt_backwards_fieldset_timevarying():
+    """test_pset_execute_outputdt_backwards() still passed despite #1722 as it doesn't account for time-varying fields,
+    which for some reason #1722
+    """
+    outputdt = timedelta(hours=1)
+    runtime = timedelta(days=2)
```
Comment (VeckoTheGecko) [inline tests/test_particlesets.py:443]: I encountered a bit of difficulty writing these tests as even here changing `runtime` to `days=1` made the bug go away.

@willirath did you also encounter some difficulty writing your minimal example?

**Review by VeckoTheGecko — APPROVED** [review]

File: tests/test_particlesets.py
Comment (erikvansebille) [inline tests/test_particlesets.py:1]: I would move the test to `tests/particlefile.py`, as it is a bug that has to do with the writing of the particleset. All output tests are in `tests/particlefile.py`

**Comment by VeckoTheGecko** [comment]
I think we can benefit with a refactor of `pset.execute()` and more rigorous testing of it. That will be a focus for me, and we can roll that into the next minor release.

We plan to do a patch release where the only difference from v3.0.4 is fixing this bug. Plan:
- Isolate this patch into new branch and create release
- Yank v3.0.4 from conda/pypi

File: tests/test_particlesets.py
Comment (VeckoTheGecko) [reply tests/test_particlesets.py:1]: done :)

File: tests/test_particlesets.py (hunk 1 above)
Comment (willirath) [reply tests/test_particlesets.py]: > @willirath did you also encounter some difficulty writing your minimal example?

No, I basically posted exactly the code that let me discover the bug. So maybe I was just "lucky".

**Pull request merged**